include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,129 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <cmath>
+#include <limits>
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+
+        /// Distributes `n >= 2` values linearly in the closed interval [`from`, `to`].
+        template<typename T>
+        struct linear_distribute_view
+          : view_facade<linear_distribute_view<T>, finite>
+        {
+            static_assert(std::is_arithmetic<T>(),
+                          "T must model the Arithmetic<T> concept; that is, "
+                          "it must be one of the native integer or floating-point types");
+        private:
+            friend range_access;
+
+            T from_, to_, spacing_;
+
+            constexpr T read() const
+            {
+                return from_;
+            }
+            RANGES_CXX14_CONSTEXPR void next()
+            {
+                from_ += spacing_;
```

Comment:
From Sean (emphasis mine):
> There are many ways to deal with the error at each step, two common ways are error accumulation and stochastic rounding. A simple method of error accumulation is to subtract off each segment **and re-divide the remaining part into the remaining number of segments**.

You're not doing it right. ;-) I think this is closer to Sean's intent:

```c++
from_ += (to_ - from_) / n_--;
```

... where `n_` is the (integral) number of segments. You're done when `n_` is zero.